Add ability to configure the apache logs and collect them

simonpasquier opened the pull request:
This change adds a new parameter to change the access log format. If
absent, the default is 'vhost_combined' as before.

Commit:
Add ability to configure the apache logs

This change supports a new entry in the model to tune the configuration
of the error and custom logs. If absent, it defaults to the same
settings as before.

fpytloun commented on README.rst:
```diff
@@ -150,6 +150,20 @@ Tune security settings (these are default):
+            foo:
+              enabled: true
+              type: static
+              custom_log_format: >-
```
Maybe this would be more generic and adding possibility to change log file location, etc. or disable at all

```
log:
  custom:
    file: /var/log/apache2/mylittleponysitecustom.log
    format: ...
  error:
    enabled: false
```

simonpasquier replied:
I thought about this but was too lazy to go through it ;-)
I'll amend the PR based on your comment.

fpytloun replied:
Awesome :-)

Commit:
Collect Keystone WSGI logs

Keystone can run as a WSGI application under Apache and the access logs
should be collected.

simonpasquier commented:
@fpytloun @cznewt can you check this PR? I figured out that the configuration of the log_collector service for the Keystone WSGI logs has to be done in the Apache formula because the Keystone formula can't figure out the location and format of the Apache logs.

fpytloun commented on apache/files/_log.conf:
```diff
@@ -1,5 +1,10 @@
+	ErrorLog {{ error_log.file|default('/var/log/apache2/' ~ site_name ~ '.error.log') }}
+{%- endif %}
+{%- if custom_log.get('enabled', True) %}
+	CustomLog {{ custom_log.file|default('/var/log/apache2/' ~ site_name ~ '.access.log') }} "{{ custom_log.format|default('vhost_combined') }}"
```
You should maybe take log directory path from Jinja, because AFAIK on RHEL it's /var/log/httpd or something?
Also please retab this file to have 4 spaces instead of one tab.

fpytloun commented on apache/meta/heka.yml:
```diff
@@ -0,0 +1,33 @@
+
+  input:
+{%- for site_name, site in server.site.iteritems() %}
+{%- if site.type == 'keystone' %}
```
This should better go to keystone formula but I am probably fine with it.
